frontend/src/components/data-table/chart-transforms/chart-schemas.ts

```diff
@@ -1,40 +1,54 @@
 /* Copyright 2024 Marimo. All rights reserved. */
 
+/**
+ * Zod schema validation for marimo chart configuration.
+ */
+
 import { DATA_TYPES } from "@/core/kernel/messages";
-import { AGGREGATION_FNS } from "@/plugins/impl/data-frames/types";
 import { z } from "zod";
+import {
+  AGGREGATION_FNS,
+  NONE_AGGREGATION,
+  SELECTABLE_DATA_TYPES,
+  SORT_TYPES,
+  TIME_UNITS,
+} from "./types";
+import { DEFAULT_COLOR_SCHEME, EMPTY_VALUE } from "./constants";
 
-export const DEFAULT_AGGREGATION = "default";
 export const DEFAULT_BIN_VALUE = 0;
 export const NONE_GROUP_BY = "None";
-export const DEFAULT_COLOR_SCHEME = "default";
 
 export const BinSchema = z.object({
   binned: z.boolean().optional(),
   step: z.number().optional(),
 });
 
+export const AxisSchema = z
+  .object({
+    field: z.string().optional(),
+    type: z.enum([...DATA_TYPES, EMPTY_VALUE]).optional(),
+    selectedDataType: z
+      .enum([...SELECTABLE_DATA_TYPES, EMPTY_VALUE])
+      .optional(),
+    aggregate: z.enum(AGGREGATION_FNS).default(NONE_AGGREGATION).optional(),
+    sort: z.enum(SORT_TYPES).default("ascending").optional(),
+    timeUnit: z.enum(TIME_UNITS).optional(),
+  })
+  .optional();
+
 export const ChartSchema = z.object({
   general: z.object({
     title: z.string().optional(),
-    xColumn: z.object({
-      field: z.string().optional(),
-      type: z.enum(DATA_TYPES).optional(),
-    }),
-    yColumn: z.object({
-      field: z.string().optional(),
-      type: z.enum(DATA_TYPES).optional(),
-      agg: z
-        .enum([...AGGREGATION_FNS, DEFAULT_AGGREGATION])
-        .default(DEFAULT_AGGREGATION)
-        .optional(),
-    }),
+    xColumn: AxisSchema,
+    yColumn: AxisSchema,
     horizontal: z.boolean().optional(),
-    groupByColumn: z
+    colorByColumn: z
```

Comment:
could maybe reuse `AxisSchema` here too (we just don't need to expose everything)